jupytercalpoly/Jupyter-Bifrost
# Clicking axis to change variable

Related to #14 

## Changes
- [x] <del> Added BifrostVega to handle the events on vega graph.
- [x] Added BifrostVega to place the two divs on new render.
- [x] <del>  Added click event listener to Vega view and used bounding box to check if it's on the axis.
- [x] Added two divs to handle click and hover event on axes.
- [x] Highlighted the axis on hover.
- [x] Add click on axis to open the variable tab.
- [x] Add filter bar below the axis on click.
- [x] Open categorical filter on variable tab when the axis has a categorical variable.
- [x] Made graph state in sync with a data tab.
- [x] The new pill won't get active once its encoding and field are added. 
- [x] Clicking the axis when tabs beside the data tab are open would hide the filter bar on graph.
- [x] Adding a new pill would hide the filter bar on graph.

## Test
1. Plot the bifrost graph
2. Click on quantitative axis to see the filter bar
3. Click on qualitative axis to see the categorical filter on variable tab.

## Future Work
* Add div for color/ size to open the filter screen on click.
* Fold the categorical filter screen if it's open when categorical axis clicked.
